architecture/event_processor.md

```diff
@@ -0,0 +1,137 @@
+# Analytic Event Processor
+
+The Event Processor is responsible for consuming, batching, and delivering events generated
+by the server and client-side LaunchDarkly SDKs.
+
+```mermaid
+classDiagram
+    IEventProcessor <|-- NullEventProcessor
+    IEventProcessor <|-- AsioEventProcessor
+
+    AsioEventProcessor *-- LRUCache
```

Comment:
I think, in my diagrams, I had aggregate arrows the wrong direction. I think in your diagram you have associations the wrong direction.

![image](https://github.com/launchdarkly/cpp-sdks/assets/4955475/4214ac17-705a-4b20-86c0-59c4e1df48f1)
From: https://www.omg.org/spec/UML/2.5.1/PDF

Reply:
I really wish the mermaid docs went into it with a quick reference, but they don't. And I see why they don't, it's a huge task.